plugins/plugin-chart-table/src/transformProps.ts

```diff
@@ -24,14 +24,52 @@ import {
   getTimeFormatter,
   smartDateFormatter,
   getTimeFormatterForGranularity,
+  TimeFormatFunction,
+  TimeFormatter,
 } from '@superset-ui/time-format';
 
 import isEqualArray from './utils/isEqualArray';
 import { TableChartProps, TableChartTransformedProps, DataType, DataColumnMeta } from './types';
 
 const { PERCENT_3_POINT } = NumberFormats;
 const TIME_COLUMN = '__timestamp';
-const toString = (x: DataRecordValue) => String(x);
+const REGEXP_TIMESTAMP_NO_TIMEZONE = /T(\d{2}:){2}\d{2}$/;
+
+/**
+ * Extended Date object with custom formatter.
+ * By default `String(date)` returns the raw input.
+ */
+export class DateWithFormatter extends Date {
+  formatter: TimeFormatFunction;
+
+  input: DataRecordValue;
+
+  constructor(
+    input: DataRecordValue,
+    { formatter = String, forceUTC = true }: { formatter?: TimeFormatFunction; forceUTC?: boolean },
+  ) {
+    let value = input;
+
+    // force UTC time for timestamps without a timezone
+    if (forceUTC && typeof value === 'string' && REGEXP_TIMESTAMP_NO_TIMEZONE.test(value)) {
+      value = `${value}Z`;
```

Comment:
is this correct? I thought you needed to replace the `T` with a `Z` to convert the string to UTC time

Reply:
`T` is the separator between date and time. I don't think I've ever seen timestamps like `2020-07-07Z18:20:02.495`.